app/src/main/java/com/example/cheatchess/MainActivity.kt

```diff
@@ -171,4 +144,20 @@ class MainActivity : AppCompatActivity(), ChessDelegate {
             }
         }
     }
+
+    private suspend fun getStockfishResponse(mode: String): Response<StockfishEvaluation> {
+        val positionFEN = chessModel.positionToFEN()
+        val depth = "13"
+        val options = mapOf(Pair("fen", positionFEN), Pair("depth", depth), Pair("mode", mode))
+
+        return try {
+            RetrofitInstance.api.getStockfishEvaluation(options)
+        } catch (e: IOException) {
+            Log.d("MainActivity", "IOException")
+            return Response.success(404, StockfishEvaluation("errorString",false))
```

Comment:
It is good to put it in a function but why are you returning a success when it failed?